core/src/main/java/org/hisp/dhis/android/core/arch/helpers/CollectionsHelper.kt

```diff
@@ -0,0 +1,209 @@
+     *
+     * @param objects Collection with the values to put in single quotes.
+     * @return A [String[]] with the single quoted values.
+     */
+    fun withSingleQuotationMarksArray(objects: Collection<String>?): Array<String> {
+        return objects?.map { "'$it'" }?.toTypedArray() ?: emptyArray()
+    }
+
+    /**
+     * Put in single quotes a value.
+     *
+     * @param value Value to put in single quotes.
+     * @return The single quote [String] value.
+     */
+    fun withSingleQuotationMarks(value: String?): String {
+        return "'${value ?: ""}'"
+    }
+
+    /**
+     * Build a [String] with the values separated by commas.
+     *
+     * @param values Array with the values to concatenate.
+     * @return A [String] with the concatenated values.
+     */
+    @SafeVarargs
+    private fun commaSeparatedArrayValues(vararg values: String): String {
+        return commaAndSpaceSeparatedArrayValues(arrayOf(*values)).replace(", ", ",")
+    }
+
+    /**
+     * Build a [String] with the values separated by commas.
+     *
+     * @param values Collection with the values to concatenate.
+     * @return A [String] with the concatenated values.
+     */
+    fun commaSeparatedCollectionValues(values: Collection<String>): String {
+        return commaSeparatedArrayValues(*values.toTypedArray<String>())
+    }
+
+    /**
+     * Build a [String] with the uids of the objects separated by commas.
+     *
+     * @param objectsWithUid Collection with the objects to concatenate.
+     * @return A [String] with the concatenated uids.
+     */
+    fun <O : ObjectWithUidInterface> commaSeparatedUids(objectsWithUid: Collection<O>): String {
+        return commaSeparatedCollectionValues(getUids(objectsWithUid))
+    }
+
+    /**
+     * Build a [String] with the values separated by semicolons.
+     *
+     * @param values Collection with the values to concatenate.
+     * @return A [String] with the concatenated values.
+     */
+    fun semicolonSeparatedCollectionValues(values: Collection<String>): String {
+        return joinCollectionWithSeparator(values, ";")
+    }
+
+    /**
+     * Build a [String] with the values separated by custom separators.
+     *
+     * @param values    Collection with the values to concatenate.
+     * @param separator Customizable separator.
+     * @return A [String] with the concatenated values.
+     */
+    fun joinCollectionWithSeparator(values: Collection<String>, separator: String?): String {
+        return commaSeparatedCollectionValues(values).replace(",", separator!!)
+    }
+
+    /**
+     * Divide a [Collection] in a [List] of sets with a maximum size.
+     *
+     * @param originalSet The collection to divide.
+     * @param size The maximum size of the partitions.
+     * @return A [List] with the partitions of the given collection.
+     */
+    @JvmStatic
+    fun <T> setPartition(originalSet: Collection<T>, size: Int): List<MutableSet<T>> {
```

Comment:
I think Kotlin has its own functions to create partitions (chunks). Also, the return type, could it be `Set` instead of `MutableSet`?